File: Source/DriveWorks.Sdk.Examples.VbDotNet/MyProjectExtender.vb
Imports DriveWorks.Extensibility
Imports Titan.Rules.Execution

Public Class MyProjectExtender
    Inherits ProjectExtender

    ''' <remarks>
    ''' Note the allowRunDuringLoad parameter Is set to true.
    ''' This will cause DriveWorks to initialize the function And make it available to rules before the project opens.
    ''' Because the function Is initialized before the project has been fully loaded
    ''' the function might Not have access to any objects related to the project such
    ''' as form controls, variables, etc when it Is executed. This option Is only recommended
    ''' for functions that rely solely on the parameters passed to the function.
    ''' Such as this one.
    ''' </remarks>
    <Udf(True)>
    <FunctionInfo("Multiplies two numbers.", "SDK-Starter-Examples Plugin")>
    Function MultiplyTheseNumbers(<ParamInfo("Value 1", "The first value to multiply.")> ByVal val1 As Double, <ParamInfo("Value 2", "The second value to multiply.")> ByVal val2 As Double) As Double
        Return val1 * val2
    End Function

End Class

File: Source/DriveWorks.Sdk.Examples.CSharp/MyProjectExtender.cs
using DriveWorks.Extensibility;
using Titan.Rules.Execution;

namespace DriveWorks.Sdk.Examples.CSharp
{
    public class MyProjectExtender : ProjectExtender
    {
        /// <remarks>
        /// Note the allowRunDuringLoad parameter is set to true.
        /// This will cause DriveWorks to initialize the function and make it available to rules before the project opens.
        /// Because the function is initialized before the project has been fully loaded
        /// the function might not have access to any objects related to the project such
        /// as form controls, variables, etc when it is executed. This option is only recommended
        /// for functions that rely solely on the parameters or object data which does not interact with DriveWorks.
        /// Such as this one.
        /// </remarks>
        [Udf(true)]
        [FunctionInfo("Multiplies two numbers.", "SDK-Starter-Examples Plugin")]
        public double MultiplyTheseNumbers([ParamInfo("Value 1", "The first value to multiply.")] double val1, [ParamInfo("Value 2", "The second value to multiply.")] double val2)
        {
            return val1 * val2;
        }
    }
}